docs/source/demos/punched_domain.geo

```diff
@@ -0,0 +1,26 @@
+  Point(1) = {-6, 2, 0, 0.5};
```

Comment:
"punched_domain.geo" is not really a very descriptive name for the demo. Remember that one of the ways people find demos is by looking at the files in the demos directory.

Reply:
I call these files `immersed_domain.geo` and `immersed_fem.py.rst`, but I'm open to suggestions.